lib/practice_exercises.rb

```diff
@@ -1,13 +1,166 @@
+# I decided not to use this one because
+# I think `delete_at` is O(n),
+# thereby making this function O(n^2),
+# which is not very efficient.
+
+# def remove_duplicates(list) # O(n * n) => O(n^2)
+#   # exits if list is too short to have duplicates or is nil
+#   if list.length < 1 || list.nil?
+#     return list
+#   end
+#   # for every member of the list after the first,
+#   for i in 0...list.length # O(n)
+#     # if that member is the same as the member after
+#     if list[i] == list[i + 1] 
+#       # delete that member
+#       list.delete_at(i) # O(n)
+#     end
+#   end
+#   return list
+# end
+
+
+# I decided not to use this one because `delete`
+# is effectively doing a `delete_at_` within a loop,
+# so it is not more efficient than using `delete_at`.
+
+# def remove_duplicates(list) # O(n + n^2) => O(2n^2) => O(n^2)
+#   # exits if list is too short to have duplicates or is nil
+#   if list.length < 1 || list.nil?
+#     return list
+#   end
+
+#   # for every member of the list after the first,
+#   for i in 0...list.length # O(n)
+#     # if that member is the same as the member after it
+#     if list[i] == list[i + 1] 
+#       # make that member nil
+#       list[i] = nil
+#     end
+#   end
+
+#   # delete all occurences of `nil`
+#   list.delete(nil) # O(n^2)
+
+#   # return the list
+#   return list
+# end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
-end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+# Time Complexity: O(n)
+# Space Complexity: O(1)
+def remove_duplicates(list) # O(n) + (O(n) * O(1)) => O(n)
```

Comment:
This is a great solution.  You solved it in O(n) time complexity, which is awesome!